File: sdk/go/freerangenotify/presence.go
package freerangenotify

import "context"

// PresenceClient handles user presence operations.
type PresenceClient struct {
	client *Client
}

// CheckIn registers a user's presence, enabling smart delivery routing.
// If WebhookURL is provided, it overrides the user's static webhook URL.
func (p *PresenceClient) CheckIn(ctx context.Context, params CheckInParams) error {
	return p.client.do(ctx, "POST", "/presence/check-in", params, nil)
}

File: sdk/js/src/presence.ts
import { HttpClient } from './client';
import type { CheckInParams } from './types';

export class PresenceClient {
    constructor(private readonly http: HttpClient) { }

    /**
     * Register a user's presence for smart delivery routing.
     * If webhookURL is provided, it overrides the user's static webhook URL.
     */
    async checkIn(params: CheckInParams): Promise<void> {
        await this.http.request('POST', '/presence/check-in', params);
    }
}
